src/compiler/commandLineParser.ts

```diff
@@ -371,13 +371,25 @@ namespace ts {
                     parseResponseFile(s.slice(1));
                 }
                 else if (s.charCodeAt(0) === CharacterCodes.minus) {
-                    s = s.slice(s.charCodeAt(1) === CharacterCodes.minus ? 2 : 1).toLowerCase();
+                    s = s.slice(s.charCodeAt(1) === CharacterCodes.minus ? 2 : 1);
 
                     // Try to translate short option names to their full equivalents.
-                    if (hasProperty(shortOptionNames, s)) {
+                    if (hasProperty(shortOptionNames, s.toLowerCase())) {
                         s = shortOptionNames[s];
                     }
+                    else {

```

Comment:
Right now tsc doesn't really follow anybody's conventions.  It's more like an ad-hoc compromise between PowerShell (camelcase flags and space separators for values) and Linux (double dash flags like getopt_long uses).

Even though it's not documented, the way it's implemented allows you to go full PowerShell-style and use e.g., `-init` instead of `--init`.  This is mostly about making the getopt_long way work, too.  Is there any platform whose conventions match the `-t=es3` format?  Should I make that format allowed, too?